### PR TITLE
Improved rcl docblock

### PR DESCRIPTION
**Pull request by ahcorde**
Improved rcl docblock (\param and \return)

Signed-off-by: ahcorde <ahcorde@gmail.com>

### CODE REVIEW

File: rcl/include/rcl/validate_topic_name.h [hunk 1]
```diff
@@ -100,6 +100,9 @@ rcl_validate_topic_name(
  * \param[in] topic_name_length The number of characters in topic_name.
  *
  * \sa rcl_validate_topic_name(const char *, int *, size_t *)
+ *
+ * \param[in] topic_name_length The number of characters in topic name.
```
Comment (brawner) [inline rcl/include/rcl/validate_topic_name.h:104]: This is duplicating line 100, also where are the other arguments?

File: rcl/include/rcl/wait.h [hunk 2]
```diff
@@ -109,13 +109,15 @@ rcl_get_zero_initialized_wait_set(void);
...
  * \param[in] number_of_services non-zero size of the services set
+ * \param[in] number_of_events non-zero size of the events set
  * \param[in] context the context that the wait set should be associated with
  * \param[in] allocator the allocator to use when allocating space in the sets
  * \return `RCL_RET_OK` if the wait set is initialized successfully, or
  * \return `RCL_RET_ALREADY_INIT` if the wait set is not zero initialized, or
  * \return `RCL_RET_NOT_INIT` if the given context is invalid, or
  * \return `RCL_RET_INVALID_ARGUMENT` if any arguments are invalid, or
  * \return `RCL_RET_BAD_ALLOC` if allocating memory failed, or
+ * \return `RCL_RET_WAIT_SET_INVALID` if the wait set is invalid, or
```
Comment (brawner) [inline rcl/include/rcl/wait.h:120]: According to wait.c, this returns RCL_RET_INVALID_ARGUMENT

File: rcl/include/rcl/wait.h [hunk 3]
```diff
@@ -155,6 +157,7 @@ rcl_wait_set_init(
  * \param[inout] wait_set the wait set struct to be finalized.
  * \return `RCL_RET_OK` if the finalization was successful, or
  * \return `RCL_RET_INVALID_ARGUMENT` if any arguments are invalid, or
+ * \return `RCL_RET_WAIT_SET_INVALID` if the wait set is invalid, or
```
Comment (brawner) [inline rcl/include/rcl/wait.h:160]: Same comment

File: rcl/include/rcl/wait.h [hunk 4]
```diff
@@ -179,6 +182,7 @@ rcl_wait_set_fini(rcl_wait_set_t * wait_set);
  * \param[out] allocator the rcl_allocator_t struct to which the result is copied
  * \return `RCL_RET_OK` if the allocator was successfully retrieved, or
  * \return `RCL_RET_INVALID_ARGUMENT` if any arguments are invalid, or
+ * \return `RCL_RET_WAIT_SET_INVALID` if the wait set is invalid, or
```
Comment (brawner) [inline rcl/include/rcl/wait.h:185]: Same

File: rcl/include/rcl/context.h [hunk 5]
```diff
@@ -168,6 +168,7 @@ rcl_get_zero_initialized_context(void);
  * Lock-Free          | Yes [1]
  * <i>[1] if `atomic_is_lock_free()` returns true for `atomic_uint_least64_t`</i>
  *
+ * \param[inout] context object to be deinitialized.
```
Comment (Blast545) [inline rcl/include/rcl/context.h:171]:
```suggestion
 * \param[inout] context object to be finalized.
```

File: rcl/include/rcl/publisher.h [hunk 6]
```diff
@@ -152,8 +152,7 @@ rcl_publisher_init(
   const rcl_node_t * node,
   const rosidl_message_type_support_t * type_support,
   const char * topic_name,
-  const rcl_publisher_options_t * options
-);
+  const rcl_publisher_options_t * options);
```
Comment (Blast545) [inline rcl/include/rcl/publisher.h:155]: There are other functions in this same file with a different style, mind changing them to have the same style as this (or the previous one before this change)?

**Comment by ahcorde** [comment]
Running CI up-to `rcl` and testing only rcl. This PR is just docblock. 

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/10880/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/6240/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/8861/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/10773/)

**Pull request merged**